tools/run_examples.py

```diff
@@ -0,0 +1,190 @@
+import argparse
+import os
+from pathlib import Path
+import subprocess
+import sys
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+EXAMPLES_DIR = Path(__file__).parents[1] / Path("examples")
+ALL_EXAMPLES: List[str] = [str(p) for p in EXAMPLES_DIR.glob("**/*.py")]
+# TODO (crcrpar): Allow fastaiv2 example once https://forums.fast.ai/t/ganlearner-error-no-implementation-found-on-types-that-implement-invisibletensor/83451/7 gets resolved.
+_DEFAULT_DENY_LIST: List[str] = [
+    "fastai",
+    "hydra/simple",
+    "pytorch_distributed_",
+    "rapids_",
+    "allennlp/subsample_dataset_reader",
+]
+VER_TO_DENY_LIST: Dict[Tuple[int, int], List[str]] = {
+    (3, 6): _DEFAULT_DENY_LIST + ["botorch_"],
+    (3, 7): _DEFAULT_DENY_LIST,
+    (3, 8): _DEFAULT_DENY_LIST + ["keras_", "tensorboard_", "tensorflow_", "tfkeras_"],
+}
+
+
+def _get_distributed_examples(files: List[str]) -> List[str]:
+    return [f for f in files if "distributed" in f]
+
+
+def _get_ipynb_examples(files: Optional[List[str]] = None) -> List[str]:
+    if files is None:
```

Comment:
This else block is really needed? I don't understand when the lines are executed.

Reply:
Not needed and `files` should be `List[str]`, thanks